code/ZAS/Fire.dm

```diff
@@ -389,6 +389,23 @@ var/global/list/image/charred_overlays = list()
 		process_charred_overlay()
 	return 0
 
+/atom/proc/checkburn()
+	if(on_fire)
+		return
+	if(flammable) //if an object is not on fire, is flammable, and is in an environment with temperature above its autoignition temp & sufficient oxygen, ignite it
```

Comment:
```
if(!flammable)
    return
```
Basically if there's a single easy if statement that idents over almost the entire proc, it's better to try and flip it